### ISSUE TITLE
Translate 2019 content to French

### ISSUE THREAD

**Issue by SilentJMA** [HTTPArchive/almanac.httparchive.org#539]
These are the core templates - without which we cannot release any translated chapters. They are in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [x] base.html + authors bio - @borisschapira 
- [x] base_chapter.html - @borisschapira 
- [x] contributors.html - @borisschapira 
- [x] contributors.html 2020 - @borisschapira 
- [x] error.html - @borisschapira 
- [x] index.html - @borisschapira 
- [x] index.html 2020 - @borisschapira 
- [ ] featured_chapters.html (partially done)
- [x] table_of_contents.html @borisschapira 

These are the chapters to be translated, in rough order of popularity. They exist in the [content directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content):
- [x] javascript - @borisschapira 
- [ ] css - @nico3333fr 
- [x] seo - @AymenLoukil 
- [ ] performance -  
- [ ] http2 - 
- [x] markup -  @SilentJMA & @borisschapira
- [x] third-parties - Partie Tiers @borisschapira
- [ ] fonts - 
- [x] accessibility - @nico3333fr 
- [ ] pwa - 
- [x] cms - @JustinyAhin
- [ ] security - @nico3333fr 
- [ ] media - @borisschapira 
- [x] mobile-web - @borisschapira 
- [ ] cdn - 
- [ ] ecommerce - 
- [ ] page-weight - 
- [x] caching - @allemas
- [x] compression - @allemas
- [x] resource-hints - @borisschapira 

Additionally the following pages need translated too in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [x] methodology.html @borisschapira 
- [x] accessibility_statement.html @nico3333fr 
- [ ] base_ebook.html

There is no need to translate the chapters HTML pages are they are generated off the markdown combined with the above templates.

Please include "Makes progress on #539" in all pull requests so a link is created from the PR to this issue.

---

Common notes for writing consistency are here: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide. Feel free to edit that and/or add French-specific extras by editing this comment.

French specific extra advice:

* Web Almanac => keep it that way, even if it's in English. "The Web Almanac" => "Le Web Almanac". For other projects names, like HTTP Archive, do the same: don't translate.
* Gender-inclusive language whenever possibl. Limit the use of abbreviation, because it's not very accessible, but multiply nouns "les développeurs et développeuses". Be especially vigilant when "we" is used to talk about the people who make the Web Almanac, don't invisibleize women.
* "Products", when talking about a development tools => in French, "Solutions" (not "Produits", which is most often reserved in French for off-the-shelf products or licensed software).
* Links to French version of resources if available (I'm thinking about MDN or Wikipedia, for example)
* Microtypography (spaces, punctuation, apostrophes): if possible, apply basic rules (guillemets, spaces before ":" or "%") but not if it slows you down. We can always make a pass at the end to set everything up all at once (I know the regexes to apply).
* Web is spelled web in lowercase, and similarly internet is lower case as per our style guide.
* Third parties should be "tierce(s) partie(s)".

**Comment by bazzadp** [comment]
Greats stuff @SilentJMA - I've updated your first comment showing what needs to be done. Feel free to stick your name against an item, clone the repo, and submit pull requests (PRs) with your translations.

Note a lot of the visualisation graphics are due to change in https://github.com/HTTPArchive/almanac.httparchive.org/pull/511 so might want to hold off for that for some of the chapters, but the JavaScript, Accessibility and HTML chapters are good to go now.

As I mentioned @AymenLoukil also wanted to help out with the French so if the two of you could review each others chapters that would be great. Unfortunately my French is pretty poor, but happy to give PRs a once over for other technical issues.

**Comment by SilentJMA** [comment]
@bazzadp i will use Gitlocalize

**Comment by bazzadp** [comment]
> @bazzadp i will use Gitlocalize

As per #38 (in case you didn't see this, or for others coming here late) they've just been submitting manual PRs for this instead of using Gitlocalize. Sorry not familiar enough with it gitlocalise to advise any better so probably best you do that too then.

**Comment by SilentJMA** [comment]
> > @bazzadp i will use Gitlocalize
> 
> As per #38 (in case you didn't see this, or for others coming here late) they've just been submitting manual PRs for this instead of using Gitlocalize. Sorry not familiar enough with it gitlocalise to advise any better so probably best you do that too then.

i'm working on markup maybe i could finish it tonight so i can added it

**Comment by bazzadp** [comment]
Could you also check you have the latest markdown down for the markup chapter (man that's confusing to type). We just pushed a big fix to all the chapter visuals as per #542.

Changes are here: https://github.com/HTTPArchive/almanac.httparchive.org/commit/8470bed51a02201781176954f56b6103ac392f6c#diff-2f5abb57ff92adb7644369369d209f1e

**Comment by SilentJMA** [comment]
> Could you also check you have the latest markdown down for the markup chapter (man that's confusing to type). We just pushed a big fix to all the chapter visuals as per #542.
> 
> Changes are here: [8470bed#diff-2f5abb57ff92adb7644369369d209f1e](https://github.com/HTTPArchive/almanac.httparchive.org/commit/8470bed51a02201781176954f56b6103ac392f6c#diff-2f5abb57ff92adb7644369369d209f1e)

when i will finish the markup file i will go back and edited the change

**Comment by SilentJMA** [comment]
> > Could you also check you have the latest markdown down for the markup chapter (man that's confusing to type). We just pushed a big fix to all the chapter visuals as per #542.
> > Changes are here: [8470bed#diff-2f5abb57ff92adb7644369369d209f1e](https://github.com/HTTPArchive/almanac.httparchive.org/commit/8470bed51a02201781176954f56b6103ac392f6c#diff-2f5abb57ff92adb7644369369d209f1e)
> 
> when i will finish the markup file i will go back and edited the change

is it okay to add the file not finished to edit the change ?

**Comment by bazzadp** [comment]
> is it okay to add the file not finished to edit the change ?

Sorry not sure what you mean? You want to submit the file partially translated? Or with the old chapter visuals? If so, then yes it's possible, but personally I would do all or nothing if possible. One less thing to track!

You can do several commits, and push to your local branch, if you're worried about losing local commits.

But let me know what you're thinking and can maybe advise better.

**Comment by SilentJMA** [comment]
> > is it okay to add the file not finished to edit the change ?
> 
> Sorry not sure what you mean? You want to submit the file partially translated? Or with the old chapter visuals? If so, then yes it's possible, but personally I would do all or nothing if possible. One less thing to track!
> 
> You can do several commits, and push to your local branch, if you're worried about losing local commits.
> 
> But let me know what you're thinking and can maybe advise better.

because i translated the file its still just the party that got changes but its okay i will go back to you in a few minutes

**Comment by SilentJMA** [comment]
> > is it okay to add the file not finished to edit the change ?
> 
> Sorry not sure what you mean? You want to submit the file partially translated? Or with the old chapter visuals? If so, then yes it's possible, but personally I would do all or nothing if possible. One less thing to track!
> 
> You can do several commits, and push to your local branch, if you're worried about losing local commits.
> 
> But let me know what you're thinking and can maybe advise better.

markup file finished

**Comment by bazzadp** [comment]
Cool open a pull request for that and then @AymenLoukil can review when he gets a chance.

**Comment by AymenLoukil** [comment]
Yes. Please ping me when PR is open. 
I also added my name in front of some chapters.

**Comment by SilentJMA** [comment]
> Yes. Please ping me when PR is open.
> I also added my name in front of some chapters.

@AymenLoukil & @bazzadp  You can check please its really hard to translate to FR

**Comment by borisschapira** [comment]
Hi y'all. If I can help translating, I'm available. Feel free to give me a chapter and I'll PR a french translation / or I can help readproof/validate existing translations.

**Comment by bazzadp** [comment]
Excellent!

At this rate the French translation might be the first to finish - even though it only started last. Is that a challenge!?!? 😜

#550 is waiting on a review from someone with much more French than me.

Also the non-chapter template files haven’t been assigned out so can start on those if you fancy while waiting for the other two to give up some of their greedily grabbed chapters 😉 Those are good ones to get going anyway as they allow the site to be run up in development to actually see the content in a real browser!!

**Comment by bazzadp** [comment]
Also see if you can request to join the Translator team: https://github.com/orgs/HTTPArchive/teams/translators/members

**Comment by borisschapira** [comment]
@bazzadp I can't access this page, so I guess I can't request to join.

**Comment by bazzadp** [comment]
Ah man. Same with this one? https://github.com/orgs/HTTPArchive/teams/almanac-contributors/members

**Comment by borisschapira** [comment]
@bazzadp Yes. I don't even see the "Teams" tab, not being in a team.

![Capture](https://user-images.githubusercontent.com/284742/69760264-53e2dd80-1164-11ea-8df6-e678eae0b31c.png)

**Comment by borisschapira** [comment]
@SilentJMA You can assign to me the pages of the language specific templates directory, I can help by translating what needs to be translated there.

**Comment by bazzadp** [comment]
> @SilentJMA You can assign to me the pages of the language specific templates directory, I can help by translating what needs to be translated there.

Assigned the first 3 to you there. Some of them are very small and may not even have anything to translate in them, but they are the ones necessary to run the chapter pages in French (once we have at least one of them).

Feel free to comment to grab the rest of those templates once those are done.

Presume you can’t edit the first comment even though you’re assigned to it? Hopefully @rviscomi can get your access sorted soon.

**Comment by borisschapira** [comment]
@SilentJMA @bazzadp @AymenLoukil In addition, I think we need common rules for translation. Some proposals:

* Web Almanach => keep it that way, even if it's in English. "The Web Almanac" => "Le Web Almanac".
* For other projects names, 
* "Products", when talking about a development tools => "Solutions" (not "Produits", which is most often reserved in French for off-the-shelf products or licensed software).
* Links to French version of resource if available (I'm thinking about MDN, for example)
* Microtypography (spaces, punctuation): if possible, apply basic rules (guillemets, spaces before ":" or "%") but not if it slows you down. We can always make a pass at the end to set everything up all at once (I know [the regexes to apply](https://github.com/borisschapira/jekyll-microtypo/blob/master/lib/jekyll/microtypo.rb#L92)).

Feel free to add things, so that we can translate in a consistent way

**Comment by borisschapira** [comment]
French translators, FYI: if you need/want to execute the project locally to visualize the changes you're contributing, [here's what you need to change](https://github.com/borisschapira/almanac.httparchive.org/commit/5d7fca29c9211c2a2ab561beed9abc17a5300d87) (then follow ./src/README.md python instructions).

**Comment by bazzadp** [comment]
Yup. Feel free to submit that first one (to add French as a Language) as part of your PR when doing the templates.

The second one (to turn on French) we’ll hold off until we have it all done. Though we have talked about making it easier to release chapter by chapter at some point.

**Comment by SilentJMA** [comment]
@borisschapira @AymenLoukil you can join me on a team on Gitlocalize so we can work on it

**Comment by bazzadp** [comment]
@borisschapira I assigned the rest of the non-chapter pages to yourself since you stormed through the first set.

**Comment by SilentJMA** [comment]
> @borisschapira I assigned the rest of the non-chapter pages to yourself since you stormed through the first set.

I add you also @borisschapira there in GitLocalize chose whatever you want and work

**Comment by borisschapira** [comment]
> I add you also @borisschapira there in GitLocalize chose whatever you want and work

Perfect! I'll focus on the remaining templates pages tonight, and then I'll contribute.

**Comment by borisschapira** [comment]
Ok I have PR'ed all the templates that have been affected to me. Feel free to "give" me some chapters and I'll translate them as well. @SilentJMA @AymenLoukil: do you want to unburden yourself of a few chapters?

**Comment by nico3333fr** [comment]
I may help to proofread in french, just tell me which part I have to correct :)

**Comment by AymenLoukil** [comment]
> > I add you also @borisschapira there in GitLocalize chose whatever you want and work
> 
> Perfect! I'll focus on the remaining templates pages tonight, and then I'll contribute.



> Ok I have PR'ed all the templates that have been affected to me. Feel free to "give" me some chapters and I'll translate them as well. @SilentJMA @AymenLoukil: do you want to unburden yourself of a few chapters?

I unassigned myself to some chapters.

**Comment by allemas** [comment]
Hi! Hi!
I can help you to translate too! let me know if I can help you and which part to translate :)

**Comment by borisschapira** [comment]
@allemas @nico3333fr Welcome!

@nico3333fr: there are 2 PR waiting for a review, do you want readproof? In each case, you'll need to compare the FR template to its EN source if you want to check that the translation is accurate.

@allemas look at the very first post of this thread, is there one of the chapters that interests you more than the others and that you would like to translate? If so, feel free to tell us. Next step: fork `master` on you own repo, copy the markdown file of the chapter you want to translate from the [source folder](https://github.com/HTTPArchive/almanac.httparchive.org/tree/master/src/content/en/2019) into the equivalent `fr` folder and here you go.

**Comment by SilentJMA** [comment]
> > > I add you also @borisschapira there in GitLocalize chose whatever you want and work
> > 
> > 
> > Perfect! I'll focus on the remaining templates pages tonight, and then I'll contribute.
> 
> > Ok I have PR'ed all the templates that have been affected to me. Feel free to "give" me some chapters and I'll translate them as well. @SilentJMA @AymenLoukil: do you want to unburden yourself of a few chapters?
> 
> I unassigned myself to some chapters.

yes i did and i'm working now on Gitlocalize

**Comment by bazzadp** [comment]
Yes, welcome @allemas and @nico3333fr ! Great to have your help!

> yes i did and i'm working now on Gitlocalize

I'm still struggling with Gitlocalize.

Is it best for everyone to work from @SilentJMA 's GitLocalize for French? In which case can you give everyone edit access @SilentJMA ?

Or should you all work on your own forks? And potentially have your own GitLocalize there if you want?

Sorry but not familair with it to know...

**Comment by SilentJMA** [comment]
> Yes, welcome @allemas and @nico3333fr ! Great to have your help!
> 
> > yes i did and i'm working now on Gitlocalize
> 
> I'm still struggling with Gitlocalize.
> 
> Is it best for everyone to work from @SilentJMA 's GitLocalize for French? In which case can you give everyone edit access @SilentJMA ?
> 
> Or should you all work on your own forks? And potentially have your own GitLocalize there if you want?
> 
> Sorry but not familair with it to know...
> 
> We can work on my RP so we can keep track on each other



> Yes, welcome @allemas and @nico3333fr ! Great to have your help!
> 
> > yes i did and i'm working now on Gitlocalize
> 
> I'm still struggling with Gitlocalize.
> 
> Is it best for everyone to work from @SilentJMA 's GitLocalize for French? In which case can you give everyone edit access @SilentJMA ?
> 
> Or should you all work on your own forks? And potentially have your own GitLocalize there if you want?
> 
> Sorry but not familair with it to know...

We can work on my RP so we can keep track on each other

**Comment by allemas** [comment]
Thanks ! 
I forked the repo and I'd start with caching. Is it ok? :)

**Comment by SilentJMA** [comment]
> thanks to all of you!
> I forked the repo and I'd start with caching. Is it ok @borisschapira ? :)

i will add you to gitlocalize and work on it there

**Comment by SilentJMA** [comment]
> > thanks to all of you!
> > I forked the repo and I'd start with caching. Is it ok @borisschapira ? :)
> i will add you to gitlocalize and work on it there so we can keep track of each other ; create your account there by singing from github

**Comment by nico3333fr** [comment]
> @nico3333fr: there are 2 PR waiting for a review, do you want to readproof? In each case, you'll need to compare the FR template to its EN source if you want to check that the translation is accurate.

Yes, will jump on it asap :)

**Comment by bazzadp** [comment]
Hey @allemas / @nico3333fr I've just sent you an invite to the project so we can assign tasks to you. Let me know once accepted (or if you didn't get the invite!)

**Comment by allemas** [comment]
Thanks, it's ok for me now :)

**Comment by SilentJMA** [comment]
> Thanks, it's ok for me now :)

@allemas did you join gitlocalize so i can add you to the team ?

**Comment by borisschapira** [comment]
Note: I added "Common notes for writing consistency:" to the top post of this thread.

**Comment by borisschapira** [comment]
@SilentJMA I see that you've kept the "third-parties" chapter. Please note some I translated the introduction because it's featured on the index template. If you want to modify something (because it's more consistent with the way you write), please also modify [the featured extract](https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/templates/fr/2019/index.html#L99).

**Comment by allemas** [comment]
> > Thanks, it's ok for me now :)
> 
> @allemas did you join gitlocalize so i can add you to the team ?

Yes i did, what is the next step ?

**Comment by SilentJMA** [comment]
> @SilentJMA I see that you've kept the "third-parties" chapter. Please note some I translated the introduction because it's featured on the index template. If you want to modify something (because it's more consistent with the way you write), please also modify [the featured extract](https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/templates/fr/2019/index.html#L99).

i will let you get it and i will get something else

**Comment by SilentJMA** [comment]
> > > Thanks, it's ok for me now :)
> > 
> > 
> > @allemas did you join gitlocalize so i can add you to the team ?
> 
> Yes i did, what is the next step ?

done

**Comment by nico3333fr** [comment]
> Hey @allemas / @nico3333fr I've just sent you an invite to the project so we can assign tasks to you. Let me know once accepted (or if you didn't get the invite!)

Done, accepted :)

**Comment by nico3333fr** [comment]
You may assign me the accessibility chapter @SilentJMA :)

**Comment by bazzadp** [comment]
> You may assign me the accessibility chapter @SilentJMA :)

You should now have permission to edit the first comment to assign it yourself @nico3333fr

**Comment by nico3333fr** [comment]
@bazzadp right, done, thanks :)

**Comment by bazzadp** [comment]
How are we getting on guys?

I just checked our GitLocalize and see three chapters in review for a couple of weeks. Are any of them ready to be submitted as pull requests yet or still working through them?

**Comment by borisschapira** [comment]
I've been pretty busy for the last three weeks, but here I am again. I PR'ed a chapter and now I'll move on to the review of @SilentJMA's work.

**Comment by JustinyAhin** [comment]
Hi ✋.

I found about this year Web Almanac a few days ago, and saw today on a tweet of @borisschapira that it is possible to contribute with translation.

I'll start translating the CMS chapter. So, how does it works? Does it needs to be assigned to me?

Also, I'm a native French speaker, so I'll be happy to review PR :)

**Comment by bazzadp** [comment]
Hi @JustinyAhin , welcome onboard!

I've assigned you the CMS chapter by editing the [first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/539#issue-527885745). I've also sent you invites to the team so you'll be able to edit this comment in future.

The way it works is you grab one of the unassigned chapters like above, clone the repo, and then submit a PR with the translated chapter. The chapters are in mark down format in the [content directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/master/src/content) and you should copy the `en` version to the `fr` folder (the `fr` folder doesn't exist at the time of writing but should soon if #595 is approved). There are also some conventions to use in the translations in the [first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/539#issue-527885745).

Ideally you should also do an `'npm install` from the src directory, and then use `npm run generate` to generate the HTML from the markdown and include both the markdown chapter file and HTML chapter file in the PR, but we can help with this step if you're less familiar with npm. Then we need one of the other French speakers to review and approve the PR. Myself, or one of the other core members, will also give it a review for technical correctness (but our French is not good enough to review the language) and once we have the two reviews, we'll merge it to master. Please keep the markdown as similar to English version in terms of line spacing and formating as possible to assist with this technical review and any future changes we need to do.

So before you start on that CMS chapter, it would be good for you to review the translation that @borisschapira has done in #595, to get a feel for how to translate. It's primarily the markdown file that needs review, and not the HTML. Have a read and make any suggestions directly on the PR where @borisschapira can either accept your suggestion or not. If you have no suggestions and think the translation is good, then approve the PR and say that in the comment and then we can merge it.

@SilentJMA also is using [GitLocalize](https://gitlocalize.com/repo/3563) and am sure he will add you to his repo there if you prefer to use that. Or you can just submit manual PRs like @borisschapira did for #595.

And one more thing - add yourself to the 2019.json file in the Translators team once you've made your first contribution as you'll officially be part of the team and then will appear on the [contributors page](https://almanac.httparchive.org/en/2019/contributors) in the next release. [Example PR] for that (https://github.com/HTTPArchive/almanac.httparchive.org/commit/c8005c7c1641d6e659c035b02cb3289c29e3e3c1).

Hopefully that all makes sense but if not then feel free to ask any questions. We're a friendly bunch here and appreciative of the time you all are giving to help with this project so don't be afraid to ask anything at all!

**Comment by nico3333fr** [comment]
I was also very busy, but next week, I'll probably be available :)

**Comment by bazzadp** [comment]
Not a problem all. Appreciate this can be a busy time of year and this is being done on a volunteer basis. Hope you all enjoy the holiday season and don't worry about getting this this when you should be taking time off - I was just checking we hadn't missed a PR that was ready to land!

**Comment by rviscomi** [comment]
Just wanted to pop in and say that I'm super impressed to see all of the support for getting the content translated to French. Thank you to everyone who's stepping up!

**Comment by bazzadp** [comment]
@nico3333fr I added you to the [Contributors page](https://almanac.httparchive.org/fr/2019/contributors?teams=translators) in https://github.com/HTTPArchive/almanac.httparchive.org/commit/42c710325f4cf6568ad220b69dd4b0cc13d4bd3f. Hope that’s OK and feel free to submit a PR to correct or add any more details.

**Comment by nico3333fr** [comment]
@bazzadp yes, thanks a lot, sorry, with lock-down, I didn't have the time to answer earlier.

**Comment by bazzadp** [comment]
Hey all, I know there’s a lot going on in the world right now and some of you may not have the time to look at this right now, but we were making great progress on this previously, so thought no harm in reaching out with a friendly ping! Don’t worry of you can’t spare the time right now - totally understandable.

**Comment by borisschapira** [comment]
Hey @bazzadp, hope you're well. As you guessed, everything's crazy right now, but I didn't forget my involvement. As soon as I have some time alone, I'll come back to it.

**Comment by nico3333fr** [comment]
Hey :) 
Same for me, lockdown was complicated (hard to do the teacher, the work, cook, etc.) at the same time, but I hope to send a PR for accessibility soon ^^

**Comment by AymenLoukil** [comment]
Hello,
Hope you are fine.

I'm working on a new chapter translation. I faced an issue on GitLocalize on a big section saying  : Something's wrong! Please check the translation that you added and try again.
when i hit the submit button.. any idea why it is behavving like this ? 

Thanks

**Comment by borisschapira** [comment]
@AymenLoukil tbh I tried GitLocalize when working on @SilentJMA's fork but I switched to pure git/github manipulations after many errors I could not explain. If you're working on a single chapter, could you remove yourself from the other ones on this thread first post, so that we keep track on who's working on what?

**Comment by nico3333fr** [comment]
Here you go dear FR proofreaders => https://github.com/HTTPArchive/almanac.httparchive.org/pull/759 ;)

**Comment by nico3333fr** [comment]
May I be assigned to the translation of Security one ? :)

**Comment by borisschapira** [comment]
@AymenLoukil Where are you on the Performance chapter? Can I help?

**Comment by bazzadp** [comment]
> May I be assigned to the translation of Security one ? :)

Done.

@rviscomi any way to give the team edit access back to issues? Otherwise big dependency on you and me. And I might eventually get bored of this 😋

**Comment by rviscomi** [comment]
Sorry for the trouble. I've revered the change which should make issues self-assignable again. Instead, I've applied a branch-specific access rule so that only Barry or myself can merge PRs into `master`. Let me know if this becomes too restrictive.

**Comment by bazzadp** [comment]
Thanks @rviscomi !

**Comment by borisschapira** [comment]
I start to translate the Mobile Web chapter.

**Comment by nico3333fr** [comment]
I'll take also the accessibility statement, might be less difficult to translate :)

**Comment by bazzadp** [comment]
> I'll take also the accessibility statement, might be less difficult to translate :)

First time I wrote one of those, so let me know if you have suggestions for it!

**Comment by borisschapira** [comment]
> I start to translate the Mobile Web chapter.

PR proposed! #838

@AymenLoukil Up on the Performance chapter? I'd really like to continue with this one, if you're not doing it.

**Comment by AymenLoukil** [comment]
> > I start to translate the Mobile Web chapter.
> 
> PR proposed! #838
> 
> @AymenLoukil Up on the Performance chapter? I'd really like to continue with this one, if you're not doing it.

@borisschapira Yes it is in progress.

**Comment by bazzadp** [comment]
> I'll take also the accessibility statement, might be less difficult to translate :)

FYI @nico3333fr I just added to that in #832 for the accessible PDF version of the almanac.

**Comment by allemas** [comment]
Hi everyone , 
I start to translate the Compression chapter .

**Comment by bazzadp** [comment]
Bonjour team!

We seem to have lost some momentum here. Would love to get this finished out well before 2020 edition is published as after that the chances of getting it finished plummets even further!

Do any of you have any more time to work on this?

@AymenLoukil / @SilentJMA - you've quite a few chapters tagged against your name but haven't been around lately. Can you let us know if you're still planning on doing all of them? If I don't hear back then will remove your name so others can grab them if they want. I know from above, that @borisschapira was keen to tackle the performance chapter so @AymenLoukil do let us know if you made a start on that, or could hand it over to him if you don't have the time to work on it?

I know it isn't easy but with a little bit of effort am sure we could get a good few more chapters translated, and ideally them all!

**Comment by nico3333fr** [comment]
One more, done quickly :) https://github.com/HTTPArchive/almanac.httparchive.org/pull/1196 ;)

**Comment by nico3333fr** [comment]
I hope to be able to jump into security one soon.

**Comment by borisschapira** [comment]
I'll try to translate "Media" in the next few days.

**Comment by rviscomi** [comment]
Great work everyone! I think the JS and CSS chapters would be good to translate soon as those got the most page views. Should we unassign those chapters to make sure they're not blocked/stale?

**Comment by borisschapira** [comment]
I think we can, and we should. If I take JS and @nico3333fr takes CSS, they would be ready before the end of the month.

**Comment by nico3333fr** [comment]
Please update the issue and assign CSS to me ;)

**Comment by bazzadp** [comment]
> Please update the issue and assign CSS to me ;)

You should be able to edit the issue yourself. Let me know if that is not the case.

**Comment by nico3333fr** [comment]
Yes, you're right, done :)

**Comment by bazzadp** [comment]
OK if this isn’t incentive to get the translation finished then I don’t know what is!
https://twitter.com/httparchive/status/1294286600059396097?s=21

**Comment by borisschapira** [comment]
I've started the translation of the JS chapter

**Comment by borisschapira** [comment]
I'm starting the Media chapter.

**Comment by arigaud-ca** [comment]
Hi, following our discussions on Twitter with @nico3333fr  and @borisschapira I would be glad to join the team.
I saw that PWA, Page Weight or Fonts are still to translate. Do you have a preference as far as priorities are concerned?

**Comment by borisschapira** [comment]
👋  @arigaud-ca, wanna help?

**Comment by bazzadp** [comment]
Hi @arigaud-ca - welcome on board. I've also sent you invite to the team.

The chapters in the [first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/539#issue-527885745) are in rough order of popularity so go with that. Fonts is the most read out of those three chapters if particularly interested in those subjects. Once you accept that team invite you'll be able to edit the first comment and "claim" a chapter so people know you are working on it.

BTW, these pages are worth a look to get up to speed on our process:
- [Translator's Guid](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide)
- [Git Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Git-Guide), if new to Git or need a refresher

> I think I remmember there's some kind of stats page to see the chapters that are the more visited, but I can't find the URL. @rviscomi, @bazzadp would you know?

Presume you mean this one? https://datastudio.google.com/reporting/1c89f8c3-7e11-4376-8815-39966e13180e/page/1M?s=izbpWSqPUmA ?

@borisschapira looks like Performance has freed up and know you were eyeing that one!

**Comment by arigaud-ca** [comment]
Thanks @bazzadp I will spend some times to go through the documentation first.

**Comment by bazzadp** [comment]
Another gentle nudge team. Just 9 chapters to go and would love to get this closed out before 2020 chapters come along as think chances of it getting done after that will drop drastically! If you each could do one more chapter each we'd have it pretty much done!

**Comment by rviscomi** [comment]
+1 to echo @bazzadp I would love to see the 2019 Web Almanac fully supported in French! I really appreciate everyone's time and effort here so I understand if this is a lot to ask. I wish we had more translators to spread the load, or better yet, a budget to compensate you for your time.

**Comment by arigaud-ca** [comment]
Hello, I have to finish the chapter on fonts. Things have been really crazy this month, I wish October will give me a little more time.

**Comment by borisschapira** [comment]
Still have to finish the media chapter, you're right. I'll speed it up. But I don't think I'll translate another chapter for 2019, I think I'm getting a little tired of doing this, I'll take a break before 2020 chapters are published.

**Comment by nico3333fr** [comment]
This is also crazy for me (divorce, lot of work, etc.), but be sure I'll take some time to end CSS chapter!

**Comment by bazzadp** [comment]
Ah feel bad for pushing guys. Know it's a lot of effort and we all have personal lives too. No rush, and whenever you get time to work on it is entirely up to you! Thanks again for efforts so far.

**Comment by arigaud-ca** [comment]
Hi. As far as Font chapter is concerned, I have 2 last steps : check if links in the article also exists in French and check some terms to have unity between the articles.
As far as the Git process is concerned I am trying to figure out what I should do next.

**Comment by bazzadp** [comment]
Hey @arigaud-ca , once you fix those two items for Fonts, you can open a Pull Request for that to be reviewed and then merged in.

After that you can [resync your main back to upstream](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Git-Guide#resyncing-your-main-branch-to-upstream), then create a new branch and work on the next free chapter.

Looks like from [the first comment above](https://github.com/HTTPArchive/almanac.httparchive.org/issues/539#issue-527885745), that performance is next free but know @AymenLoukil was working on it and @borisschapira expressed an interest in this too if he wasn't able to complete it, so maybe http2 is better one to go for (only a slight bias from me on that one! 😁). Either way just edit [the first comment above](https://github.com/HTTPArchive/almanac.httparchive.org/issues/539#issue-527885745) to mark your name next to a free chapter that takes your fancy so others know you've "claimed" that one and then comes the hard work of the translation, and then submit a new PR for that!

That make sense? Any questions let me know.

**Comment by bazzadp** [comment]
Or do you mean you don't know how to open a pull request? Should be as simple as going to the [Pull Requests](https://github.com/HTTPArchive/almanac.httparchive.org/pulls) tab at the top of this page and GitHub should be clever enough to see you worked on something in your fork and suggest opening a pull request for that work. Again, let me know if you need any Git help and happy to guide.

**Comment by nico3333fr** [comment]
I did some progress on CSS chapter... it is really long ^^

**Comment by bazzadp** [comment]
> I did some progress on CSS chapter... it is really long ^^

It certianly is! Look how @rviscomi felt after editing it and adding the many, many figures: https://github.com/HTTPArchive/almanac.httparchive.org/pull/441

**Comment by nico3333fr** [comment]
I just noticed a bug on some FR translations 
![image](https://user-images.githubusercontent.com/2578321/99158294-8c5f2c80-26d1-11eb-8cd0-986a2c383163.png)

Is it something that can be fixed to be displayed properly or do we have to remove `&nbsp,` ? ex. https://almanac.httparchive.org/fr/2019/third-parties

**Comment by bazzadp** [comment]
Fixed in #1518 . Will see when we can release the fix.

**Comment by nico3333fr** [comment]
Fine, I won't have to fix in in future PR... :)

**Comment by bazzadp** [comment]
Though it did confuse me as not set to `&nbsp;` in the markdown and looks like a space. I presume it's not a real space but a non-braking space?

Question for you all, should we use those in future or prefer `&nbsp;` as more obvious?

**Comment by nico3333fr** [comment]
Yes, this is a non-breaking space. Personaly, I like it like that.

🎁 => CSS in french https://github.com/HTTPArchive/almanac.httparchive.org/pull/1519

**Comment by bazzadp** [comment]
Fix was released btw:

![French third-parties quote](https://user-images.githubusercontent.com/10931297/99199692-e840b880-2798-11eb-8074-bf59c049a952.png)

Thanks for reporting!

**Comment by nico3333fr** [comment]
BTW @bazzadp do you know when CSS in french will be deployed?

**Comment by bazzadp** [comment]
Just did a quick deploy there now: https://almanac.httparchive.org/fr/2019/css

**Comment by borisschapira** [comment]
I modified the title to focus this issue and 2019 translation.

For the translation of the 2020 contents, I propose myself for the coordination.

**Comment by bazzadp** [comment]
I was actually thinking of tracking both 2019 and 2020 in this issue, but with priority on 2020 content. If you look at the first comment you'll notice I've added the Capabilities chapter as ready to translate. Also if you look above you'll see I actually changed it to drop the 2019 bit from the title 😀

![Title change](https://user-images.githubusercontent.com/10931297/100861424-b4001400-3489-11eb-98bb-91d4f10c0e56.png)

Just think it will be easier to track in one issue than two open issues but do it in order of:
- Templates (done!)
- 2020 content (will add chapters as ready to translate)
- 2019 content (partially done)

Just concerned with tracking two issues (and three next year, and four the year after!) times many languages will lead to a lot of issues and people potentially working on the old content, instead of the new. Of course we could close 2019 issue and only reopen if we think people will actually get time to do it, but when why not just track in this issue?

**Comment by borisschapira** [comment]
Sorry, I didn't see your previous modifications. The way you explain it makes sense, LGTM.

**Comment by borisschapira** [comment]
I'll translate the 2020 Performance chapter!

**Comment by bazzadp** [comment]
Thought you might 😀

**Comment by nico3333fr** [comment]
I'll translate the 2020 accessibility chapter :D

**Comment by nico3333fr** [comment]
BTW @bazzadp : we do agree that all 2020 chapters referenced in this issue are the ones we can go, they are finalised, right?

**Comment by bazzadp** [comment]
Correct! Though am sure there are still typos to be found. But translators are extra good at spotting them 😁

**Comment by bazzadp** [comment]
Oh and still some discussion on Privacy in #1760 so it might change, so will remove that one.

**Comment by svgeesus** [comment]
When the CSS 2020 chapter is translated to French, I'm happy to review it. I'm not a native speaker but lived there for many years and have a good level in reading French.

**Comment by borisschapira** [comment]
Noted! Thanks @svgeesus

**Comment by nico3333fr** [comment]
I'll probably translate it, as I did for the CSS chapter of 2019 :)

**Comment by bazzadp** [comment]
Would any of the French translators be able to translate the 2020 Forward from `2020/base.html`? Quite small but notice it's a TODO in the French version.

**Comment by borisschapira** [comment]
@bazzadp I'm on it.

**Comment by stephanebachelier** [comment]
I propose to translate the Jamstack chapter in French as a first contribution.

**Comment by bazzadp** [comment]
> I propose to translate the Jamstack chapter in French as a first contribution.

Hi @stephanebachelier , there's going to be some changes to this chapter (see #1736 ) so this chapter is not ready for translation yet, which is why it's not listed in the first comment on this issue. That one is not on my list to sort, so hoping to look at it tomorrow or this weekend.

In the meantime, I've invited you to the team so if you want you can grab one of the other chapters and put your name about it in the first comment? I'll update here once Jamstack chapter is ready for translation.

**Comment by stephanebachelier** [comment]
Hi @bazzadp. I'll translate the fonts chapter. I've claimed it as you said.

**Comment by borisschapira** [comment]
@stephanebachelier If you need help, feel free to ask here or in my Twitter DMs (@borisschapira). And if you don't, I'll be happy to review your translation once you have made your PR.

**Comment by bazzadp** [comment]
Oh that reminds me. The changes to the Jamstack chapter have now been merged. So if you’ve not started fonts yet and still want to do that one, then it’s available now.

**Comment by borisschapira** [comment]
Nice! I'm interested but I'll let @stephanebachelier translate it if he wants. I start the SEO chapter instead.

**Comment by bazzadp** [comment]
Final two chapters (privacy and caching) have PRs open for them which should be merged in next week. After that we’ve all the chapters available for you all to choose from!

**Comment by borisschapira** [comment]
Welcome @excellenseo! If you want to wait for the 2021 content, so be it, but if you'd like to start with some content from 2019 or 2020 to translate in French, we'll be happy to onboard you!

**Comment by excellenseo** [comment]
Thanks @borisschapira. If there is still 2019 and 2020 content to translate, wherever I can help, I will. Just let me know.

**Comment by borisschapira** [comment]
The first post in this thread is a summary of our current situation. Each box that is not checked is a chapter to be translated. When there is a name on the right, someone is more or less actively working on it (if you want to participate, you can ask them). When there is none, then the chapter is open for translation.

**Comment by tunetheweb** [comment]
We added a small note to our Accessibility Statement about colour contrast issues:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/14f85122c88d14ee1f8dd0474479424338766625/src/templates/fr/accessibility_statement.html#L69-L71

Would be great if someone could translate this (nudge, nudge @borisschapira since I know you're very keen on Accessibility!)
